engine/lib/users.php

```diff
@@ -449,10 +449,16 @@ function user_avatar_hook($hook, $entity_type, $returnvalue, $params) {
 	$size = $params['size'];
 
 	if (isset($user->icontime)) {
-		return "avatar/view/$user->username/$size/$user->icontime";
```

Comment:
This doesn't seem too tough: https://github.com/Elgg/Elgg/issues/9469#issuecomment-194861328